soc: nxp: mcxn: configure CPU1 TrustZone access level

DerekSnell opened the pull request:
Configures AHBSC MASTER_SEC_LEVEL register for the cpu1 before cpu1 is enabled.  By default, this gives CPU1 secure and privileged access to the rest of the SOC, same as CPU0.

This also reverts feb966df36cbb8b6d847b9f0873bc79195e7fd22.  Configuring the GPIO access permissions is no longer required for CPU1 to access the GPIO.

Resolves #88965 .

Tested using the blinky sample running on CPU1 using the following commands:
```
west build -b frdm_mcxn947//cpu1 samples/basic/blinky -d blinky_cpu1 -p -- -DCONFIG_SECOND_CORE_MCUX=y
west flash -d blinky_cpu1
west build -b frdm_mcxn947//cpu0 samples/hello_world/  -d hello_cpu0_loader -p -- -DCONFIG_SECOND_CORE_MCUX=y
west flash -d hello_cpu0_loader
```
GPIO also accessible from CPU0, tested with sample blinky and button.

Commit:
Revert "boards: nxp: frdm_mcxn947: enable GPIO in all modes"

This reverts commit feb966df36cbb8b6d847b9f0873bc79195e7fd22.  That
change makes GPIO registers inaccessible from cpu0.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

DerekSnell commented:
@alxrey and @ubieda , FYI

ubieda reviewed (dismissed):
I just tested and confirmed it fixes the regression. Thanks!

alxrey commented on soc/nxp/mcx/mcxn/soc.c:
```diff
@@ -44,6 +44,11 @@ DT_FOREACH_STATUS_OKAY(nxp_lpspi, FLEXCOMM_CHECK)
+	/* Configure CPU1 TrustZone access level before CPU1 is enabled */
+	AHBSC->MASTER_SEC_LEVEL |=
+		AHBSC_MASTER_SEC_LEVEL_CPU1(CONFIG_SECOND_CORE_MCUX_ACCESS_LEVEL);
+	AHBSC->MASTER_SEC_ANTI_POL_REG = (~AHBSC->MASTER_SEC_LEVEL & 0x3FFFFFF3U) | 0x80000000U;
```
Bits 3-2 (CPU1 field) are cleared by the mask `& 0x3FFFFFF3U`, which doesn't seem to be the intended behavior.

If COP1 in `MASTER_SEC_LEVEL` is set to 1 or 2 (_Non-secure and privileged Master_, respectively _Secure and non-privileged Master_), `MASTER_SEC_ANTI_POL_REG` will not have the inverse polarity of `MASTER_SEC_LEVEL` as specified.

IMHO, the mask should be set to `0x3FFFFFFFU`.

alxrey replied:
Shouldn't macros be used instead of hard-coded values?

Concretely:
```C
AHBSC->MASTER_SEC_ANTI_POL_REG = (~AHBSC->MASTER_SEC_LEVEL & ~AHBSC_MASTER_SEC_ANTI_POL_REG_MASTER_SEC_LEVEL_ANTIPOL_LOCK_MASK) | AHBSC_MASTER_SEC_ANTI_POL_REG_MASTER_SEC_LEVEL_ANTIPOL_LOCK(2);
```

DerekSnell replied:
Great feedback, @alxrey .  Updated as you requested.  Thanks for reviewing

Commit:
soc: nxp: mcxn: configure CPU1 TrustZone access level

Configures AHBSC MASTER_SEC_LEVEL register for the cpu1 before cpu1 is
enabled.  By default, this gives CPU1 secure and privileged access to
the rest of the SOC, same as CPU0.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

ubieda reviewed (approved):
Refreshing my +1

dleach02 commented:
@alxrey did your request get satisfied?

alxrey commented:
@dleach02 Yes, it looks perfect now.

DerekSnell commented:
> @dleach02 Yes, it looks perfect now.

Thanks @alxrey , you still have a block on this PR.  Can you approve to dismiss your block?  Thank you for reviewing and your feedback.

DerekSnell commented on boards/nxp/frdm_mcxn947/board.c:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2024-2025 NXP
+ * Copyright 2024  NXP
```
Hi @dleach02 , good question
I reverted my previous commit that modified this file.  After this revert, this file was effectively not updated in 2025.  Do you think we should update the copyright year anyway?
